components/brave_wallet/renderer/eth_request_parser_unittest.cc

```diff
@@ -0,0 +1,107 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <memory>
+#include <utility>
+#include <vector>
+
+#include "brave/components/brave_wallet/renderer/eth_request_parser.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_wallet {
+
+TEST(EthRequestParserUnitTest, ParseEthSendTransactionParams) {
+  std::string json(
+      R"({
+        "params": [{
+          "from": "0x7f84E0DfF3ffd0af78770cF86c1b1DdFF99d51C8",
+          "to": "0x7f84E0DfF3ffd0af78770cF86c1b1DdFF99d51C7",
+          "gas": "0x146",
+          "gasPrice": "0x123",
+          "value": "0x25F38E9E0000000",
+          "data": "0x010203"
+        }]
+      })");
+  std::string from;
+  mojom::TxDataPtr tx_data = ParseEthSendTransactionParams(json, &from);
+  ASSERT_TRUE(!!tx_data);
+  CHECK_EQ(from, "0x7f84E0DfF3ffd0af78770cF86c1b1DdFF99d51C8");
+  CHECK_EQ(tx_data->to, "0x7f84E0DfF3ffd0af78770cF86c1b1DdFF99d51C7");
+  CHECK_EQ(tx_data->gas_limit, "0x146");
+  CHECK_EQ(tx_data->gas_price, "0x123");
+  CHECK_EQ(tx_data->value, "0x25F38E9E0000000");
+  ASSERT_EQ(tx_data->data, (std::vector<uint8_t>{1, 2, 3}));
```

Comment:
It is better to use `ASSERT_EQ` instead of `CHECK_EQ` in test. And in this case, we should use `EXPECT_EQ`

Reply:
oops nice catch, I meant `EXPECT_EQ`. Fixed and interactive rebased in b798a7fbb264fd1ff08ad10f3e43520a118952d4